src/main/resources/com/ncc/aif/restricted_hypothesis_aif.shacl

```diff
@@ -25,6 +25,59 @@ aida:SystemShape
         """ ;
     ] .
 
+#########################.;
+# 4.3 TA3 #4 Each hypothesis graph must have exactly one hypothesis importance value
+#------------------------
+aida:HypothesisImportanceRequiredShape
+    a sh:PropertyShape ;
+    sh:path aida:importance ;
+    sh:or (
+        [sh:propertyShape aida:HypothesisShape]
+        [sh:propertyShape aida:EventArgumentShape]
+        [sh:propertyShape aida:RelationArgumentShape]
+    ) ;
+    sh:minCount 1 ;
+    sh:maxCount 1 .
+
+aida:HypothesisShape
+    sh:property aida:HypothesisImportanceRequiredShape .
+
+#########################
+# 4.3 TA3 #5 Each event or relation (cluster) in the hypothesis must have exactly one event/relation importance value
+#------------------------
+aida:HypothesisEventRelationClusterImportanceRequiredShape
+    a sh:NodeShape ;
+
+    sh:target [
+        a sh:SPARQLTarget ;
+        sh:select """
+            PREFIX aida:  <https://tac.nist.gov/tracks/SM-KBP/2018/ontologies/InterchangeOntology#>
+            SELECT ?prototype
+            WHERE {
+                ?prototype a aida:SameAsCluster .
+                ?prototype aida:prototype/a ?type .
+                FILTER( ?type = aida:Event || ?type = aida:Relation )
```

Comment:
I was concerned that there would be more overhead with at FILTER than with a UNION. Let me know what you think about this alternative:
`{ ?prototype aida:prototype/a aida:Event } UNION { ?prototype aida:prototype/a aida:Relation }`

Reply:
I don't know what is more efficient with respect to SPARQL. I can update it to UNION if you feel it may be a bit faster, just let me know.

Reply:
I really don't know. I was kinda hoping you could lend some insight from your query/SPARQL work with NIST. That being said, I think we can leave it the way it is. Perhaps the 'right way' will become evident when we finish our performance analysis.